Sources/TuistCache/ContentHashing/CacheContentHasher.swift

```diff
@@ -0,0 +1,43 @@
+import Foundation
+import Basic
+
+/// CacheContentHasher
+/// It's a wrapper on top of ContentHasher that adds an in-memory cache to avoid recalculating the same hashes
+public final class CacheContentHasher: ContentHashing {
+    private let contentHasher: ContentHashing
+
+    /// In memory cache for files that have already been hashed
+    private var hashesCache: [String: String] = [:]
```

Comment:
Bear in mind that hashing a string `/path/MyFile.swift` and hashing the file at `/path/MyFile.swift` are 2 different things and therefore return 2 different results. By using `String` as the key you are treating both the same.
I'd only cache file hashes because they are the most expensive *(require IO)*.

Reply:
Update to only cache files